### PR TITLE
Add search route for shows

### PR DESCRIPTION
**Pull request by hinosxz**
## What does this PR do?
Opens a `/search` route to search for a show using Trakt API. Filtering is applied with the query parameter `name`. 

## Motivation (link to Trello card)
https://trello.com/c/7H3XU94k

## Testing Guidelines
Hit https://chowchow-staging.herokuapp.com/search?name=handmaid and you should be able to find `The Handmaid's Tale`

## Release Checklist

-   [ ] This code has automated tests
-   [x] This code has documentation
-   [x] I have a plan to verify that these changes are working as expected after deploy

### CODE REVIEW

File: src/main/java/com/centralesupelec/chowchow/search/controllers/TraktSearch.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package com.centralesupelec.chowchow.search.controllers;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (DonaCrio) [inline src/main/java/com/centralesupelec/chowchow/search/controllers/TraktSearch.java]: It works well that way, I'd just like to be consistent with the way the DTOs are written (see @ShowDTO)

File: src/main/java/com/centralesupelec/chowchow/search/web/SearchWebController.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+public class SearchWebController {
+
+    private final SearchController searchController;
+
+    @Autowired
+    public SearchWebController(SearchController searchController) {
+        this.searchController = searchController;
+    }
+
+    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (DonaCrio) [inline src/main/java/com/centralesupelec/chowchow/search/web/SearchWebController.java]: I'd have add the path, even if it's `""`, just to keep things clear.

File: src/main/java/com/centralesupelec/chowchow/search/controllers/TraktSearch.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+class Ids {
+    private int trakt;
+
+    public int getTrakt() {
+        return trakt;
+    }
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+class TraktShow {
```
Comment (DonaCrio) [inline src/main/java/com/centralesupelec/chowchow/search/controllers/TraktSearch.java]: This is definitively a DTO and it's used for the same purpose so I suggest we could rename it `TraktShowDTO` and use the same decorators and set the attributes to `final`.

File: src/main/java/com/centralesupelec/chowchow/search/controllers/TraktSearch.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return getIds().getTrakt() + ": " + getTitle();
+    }
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class TraktSearch {
```
Comment (DonaCrio) [inline src/main/java/com/centralesupelec/chowchow/search/controllers/TraktSearch.java]: Same here

File: src/main/java/com/centralesupelec/chowchow/search/web/SearchWebController.java (hunk 2 above)
Comment (hinosxz) [reply src/main/java/com/centralesupelec/chowchow/search/web/SearchWebController.java]: I find it clearer to not expose an empty path 🤷‍♂ I guess it depends, let's keep it like that if you don't mind.

**Pull request merged**